src/libcrun/linux.c

```diff
@@ -2873,17 +2873,17 @@ libcrun_reopen_dev_null (libcrun_error_t *err)
 }
 
 static int
-uidgidmap_helper (char *helper, pid_t pid, char *map_file, libcrun_error_t *err)
+uidgidmap_helper (char *helper, pid_t pid, const char *map_file, libcrun_error_t *err)
 {
 #define MAX_ARGS 20
   char pid_fmt[16];
   char *args[MAX_ARGS + 1];
-  char *next;
+  cleanup_free char *next = NULL;
```

Comment:
next is modified by the while loop, so the `free(next);` is called on the wrong pointer.

You need to use a different variable to hold the xstrdup result and make sure you don't change its value